notification/slack/channel.go

```diff
@@ -55,21 +56,33 @@ type Channel struct {
 	Name string
 }
 
-type slackError string
+type apiError struct {
+	msg    string
+	header http.Header
+}
+
+func (err apiError) Error() string { return err.msg }
+
+func mapError(ctx context.Context, err error) error {
+	var apiError *apiError
+	if !errors.As(err, &apiError) {
+		return err
+	}
 
-func (err slackError) Error() string     { return string(err) }
-func (err slackError) ClientError() bool { return true }
-func wrapError(errMsg, details string) error {
-	switch errMsg {
+	switch apiError.msg {
 	case "missing_scope":
-		// happens if the ID is for a user
-		return validation.NewFieldError("ChannelID", "Only channels supported.")
+		acceptedScopes := apiError.header.Get("X-Accepted-Oauth-Scopes")
+		providedScopes := apiError.header.Get("X-Oauth-Scopes")
+		log.Log(ctx, fmt.Errorf("Slack: missing_scope; need one of %v but got %v", acceptedScopes, providedScopes))
+		return validation.NewFieldError("ChannelID", "Permission Denied.")
 	case "channel_not_found":
 		return validation.NewFieldError("ChannelID", "Invalid Slack channel ID.")
 	case "invalid_auth", "account_inactive", "token_revoked", "not_authed":
-		return slackError("User account must be linked.")
+		log.Log(ctx, fmt.Errorf("unable to authenticate to Slack: %v", apiError.msg))
```

Comment:
This one we can just log the original, as it will already contain the context of which API call we attempted, as well as the specific error like `token_revoked`.

```suggestion
		log.Log(ctx, err)
```